src/app/jscaip/tests/Coord3D.spec.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable max-lines-per-function */
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { JSONValue } from 'src/app/utils/utils';
+import { Coord3D } from '../Coord3D';
+
+describe('Coord3D:', () => {
+
+    it('should encode optional and decode should be symetrical', () => {
+        const initialValues: MGPOptional<Coord3D>[] = [
+            MGPOptional.empty(),
+            MGPOptional.of(new Coord3D(0, 0, 0)),
+        ];
+        const encodedValues: JSONValue[] = initialValues.map(Coord3D.optionalEncoder.encode);
+        const decodedValues: MGPOptional<Coord3D>[] = encodedValues.map(Coord3D.optionalEncoder.decode);
+        const expectedValues: JSONValue[] = [
+            null,
+            { x: 0, y: 0, z: 0 },
+        ];
+        expect(encodedValues).toEqual(expectedValues);
+        expect(decodedValues).toEqual(initialValues);
+    });
+
+    it('should override equals correctly', () => {
```

Comment:
ça y'est on a plus de correctly pour les encodeurs et maintenant on a correctly pour equals ET compare (plus bas dans le même fichier) ?!

Reply:
On a 57 correctlys hahaha :D !

Reply:
qu'en fait on des nouveaux ?

Reply:
dur à dire, les equals soit on les splits et c'est facile d'être trop verbeux "should find unequals when second field are unequal" etc mais ... mais meh, mémé.
Sinon c'est pas simple :D